core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -621,14 +621,17 @@ object SparkSubmit {
     if (isKubernetesCluster) {
       childMainClass = "org.apache.spark.deploy.kubernetes.submit.Client"
       if (args.isPython) {
-        childArgs += args.primaryResource
-        childArgs += "org.apache.spark.deploy.PythonRunner"
-        childArgs += args.pyFiles
+        childArgs ++= Array("--primary-py-file", args.primaryResource)
```

Comment:
In `Client.scala` we parse the arguments array into a case class and report on errors when fields are missing.